src/pygama/raw/orca/orca_digitizers.py

```diff
@@ -210,3 +204,228 @@ def decode_packet(self, packet, packet_id, rbl):
 
         evt_rbkd[ccc].loc += 1
         return evt_rbkd[ccc].is_full()
+
+
+class ORSIS3316WaveformDecoder(OrcaDecoder):
+    """Decoder for `Struck SIS3316 <https://www.struck.de/sis3316.html>`_
+    digitizer data written by ORCA."""
+
+    def __init__(self, header: OrcaHeader = None, **kwargs) -> None:
+
+        # store an entry for every event
+        self.decoded_values_template = {
+            "packet_id": {
+                "dtype": "uint32",
+            },
+            "energy": {
+                "dtype": "uint32",
+                "units": "adc",
+            },
+            "energy_first": {
+                "dtype": "uint32",
+            },
+            "timestamp": {
+                "dtype": "uint64",
+                "units": "clock_ticks",
+            },
+            "crate": {
+                "dtype": "uint8",
+            },
+            "card": {
+                "dtype": "uint8",
+            },
+            "channel": {
+                "dtype": "uint8",
+            },
+            # waveform data
+            "waveform": {
+                "dtype": "uint16",
+                "datatype": "waveform",
+                "wf_len": 65532,  # max value. override this before initializing buffers to save RAM
+                "dt": 8,  # override if a different clock rate is used
+                "dt_units": "ns",
+                "t0_units": "ns",
+            },
+        }
+
+        self.decoded_values = {}
+        self.skipped_channels = {}
+        super().__init__(
+            header=header, **kwargs
+        )  # also initializes the garbage df (whatever that means...)
+
+    def set_header(self, header: OrcaHeader) -> None:
+        self.header = header
+
+        self.decoded_values = copy.deepcopy(self.decoded_values_template)
```

Comment:
this line seems like a bug. It will create a dec vals dict at self.decoded_values that is later appended at self.decoded_values[ccc]

I think this line can just be removed?